pkg/executor/service.go

```diff
@@ -93,17 +94,18 @@ func CreatePodService(
 	cr *kdv1.KubeDirectorCluster,
 	role *kdv1.Role,
 	podName string,
+	client k8sclient.Client,
 ) (*v1.Service, error) {
 
 	serviceType := serviceType(*cr.Spec.ServiceType)
 
-	portInfoList, portsErr := catalog.PortsForRole(cr, role.Name)
+	portInfoList, portsErr := catalog.PortsForRole(cr, role.Name, client)
 	if portsErr != nil {
 		return nil, portsErr
 	}
-	if len(portInfoList) == 0 {
-		return nil, nil
-	}
+    if len(portInfoList) == 0 {
+        return nil, nil
+    }
```

Comment:
Indentation. Running "make format" should catch this kind of thing.

Reply:
Done.